homeassistant/components/person/__init__.py

```diff
@@ -373,18 +377,34 @@ def _async_handle_tracker_update(self, entity, old_state, new_state):
         """Handle the device tracker state changes."""
         self._update_state()
 
+    def _get_latest(self, prev, curr):
+        return curr \
+               if prev is None or curr.last_updated > prev.last_updated \
+               else prev
+
     @callback
     def _update_state(self):
         """Update the state."""
-        latest = None
+        latest_home = latest_not_home = latest_gps = latest = None
         for entity_id in self._config.get(CONF_DEVICE_TRACKERS, []):
             state = self.hass.states.get(entity_id)
 
             if not state or state.state in IGNORE_STATES:
                 continue
 
-            if latest is None or state.last_updated > latest.last_updated:
-                latest = state
+            if state.attributes.get(ATTR_SOURCE_TYPE) == SOURCE_TYPE_GPS:
+                latest_gps = self._get_latest(latest_gps, state)
+            elif state.state == STATE_HOME:
+                latest_home = self._get_latest(latest_home, state)
```

Comment:
Shouldn't we call this `latest_non_gps_home` or something? Since we're excluding GPS from this group.

Reply:
Done in #21528